### PR TITLE
add changelog for release v0.5.0

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
- add changelog for release v0.5.0

#### Ticket Link
n/a

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,24 +1,118 @@
+# v0.5.0
+
+## Enhancements
+
+* Add API for fetching Fulcio configuration (https://github.com/sigstore/fulcio/pull/608)
+* Final challenge result removal 🎉  (https://github.com/sigstore/fulcio/pull/626)
```
Comment (haydentherapper) [inline CHANGELOG.md:6]: Can we group the principal refactors into one line with multiple PRs? It'd be Final challenge result removal, Move email principal to package, Move kubernetes principal to package, Move spiffe principal to package, Move github principal to its own package, Use principal in CA abstraction, Refactor x509 extension embedding logic, Add new Issuer and Principal abstractions

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,24 +1,118 @@
...
+* Move kubernetes principal to package (https://github.com/sigstore/fulcio/pull/619)
+* Move spiffe principal to package (https://github.com/sigstore/fulcio/pull/604)
+* Split pkg/server from pkg/api (https://github.com/sigstore/fulcio/pull/616)
+* Move github principal to its own package (https://github.com/sigstore/fulcio/pull/599)
+* Restict issuer claim mapping to email issuers (https://github.com/sigstore/fulcio/pull/606)
+* Move domain validation checks for URI/Username to service startup (https://github.com/sigstore/fulcio/pull/590)
+* Validate SPIFFE IDs and trust domains via library (https://github.com/sigstore/fulcio/pull/592)
+* Refactor challenge verification (https://github.com/sigstore/fulcio/pull/580)
+* identity: improve the documentation for Principal.Name() (https://github.com/sigstore/fulcio/pull/579)
+* challenges: remove ParseCSR (https://github.com/sigstore/fulcio/pull/578)
```
Comment (haydentherapper) [inline CHANGELOG.md:18]: Can we call this misc or maybe change the bug fix header to bug fix/misc? This was just switching to a sigstore/sigstore method.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,24 +1,118 @@
...
+* Use principal in CA abstraction (https://github.com/sigstore/fulcio/pull/570)
+* googleca: Don't log all identities (https://github.com/sigstore/fulcio/pull/577)
+* Small `ca` refactor (https://github.com/sigstore/fulcio/pull/569)
+* Use GenerateSerialNumber from cryptoutils (https://github.com/sigstore/fulcio/pull/571)
```
Comment (haydentherapper) [inline CHANGELOG.md:22]: misc too

**Review by haydentherapper — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (cpanato) [reply CHANGELOG.md:22]: thanks for your feedback and review @haydentherapper 
PTAL

**Pull request merged**